### ISSUE TITLE
"Select a folio" leads to 404s in Bower segment

### ISSUE THREAD

**Issue by annamorphism** [DDMAL/CantusDB#1049]
While investigating some of the Bower sources I tried to choose a folio from https://cantusdatabase.org/source/123779 from the "Select a folio" dropdown, which led to 
https://cantusdatabase.org/chants/?source=123779&folio=p.+468+%5B234v%5D
and gives a 404.
The same happens on various other Bower sources I tried, so I think it's a problem throughout the segment.

**Comment by jacobdgm** [comment]
Most folios in the database are three numbers plus one letter, but in this MS we've got folios like `p. 605 [303v]`, i.e. we have to deal with a `.`, a ` `, and `[]`s. This seems to be the problem.

Already, in the URL, we have `%5B` and `%5D` substituting for the `[` and `]`. Doing some initial experimentation, simply swapping in `%30` for ` ` and `%2E` for `.` does not cause the page to load properly. I/we'll have to do some thinking as to how to approach this. It would be worth doing some looking into what other unexpected characters show up in the `folio` field.

Nice catch, @annamorphism!

**Comment by jacobdgm** [comment]
poking around OldCantus, this feature for this source is broken in a different way: when you select a folio from the dropdown on https://cantus.uwaterloo.ca/source/123779, the page loads, but no chants are displayed: https://cantus.uwaterloo.ca/chants?source=123779&folio=p.%20470%20[235v]. Even for folios that include fewer exotic characters (https://cantus.uwaterloo.ca/chants?source=123779&folio=p.%20434), no chants are displayed. I wonder whether (for OldCantus), this is a Chants vs. Sequences thing, i.e. it's displaying all the Chants on the folio, but there are only Sequences and no Chants, so no Chants are shown...

**Comment by jacobdgm** [comment]
Having looked into this a bit, my current line of thinking is that we should simply not display the "Select a folio" or "Select a feast" dropdowns for sources that aren't part of the Cantus Segment. Main reason:
- Selecting something from the dropdowns brings you to the Browse Chants page. But since sources in the Bower Segment contain only Sequences (i.e., they contain no Chants), even if the query parameters worked properly, this page will always be empty.
  - This page is thus also broken on OldCantus - while it doesn't 404, the page it brings you to is always useless because Bower sources contain no chants

And a secondary observation:
- Doing a search for Sequences with associated feasts in my local database, I see that there exists no Sequence that has a feast. So the "Select a feast" dropdown will always be useless for sources in the Bower segment.

In summary, for sources in the Bower segment, neither of these dropdowns has ever really worked, neither on OldCantus or on NewCantus. The most elegant way to handle this issue is to simply remove these dropdowns from the Detail pages of sources in the Bower segment. @annamorphism, do you think this is reasonable?

**Comment by annamorphism** [comment]
Taking away the feasts dropdown seems like an obvious fix--in this segment the feast as recorded as stating in the manuscript (within the "rubrics" field) and not associated with a feast code, so there's no need for the dropdown either. 
 
As for the select a folio page--this seems like it _could_ be useful, but if it is going to be annoying to fix I could live without it, since it didn't really work before anyway. (As fixes to the Bower segment go, the ability to search the "rubrics" and "notes" field seems way more useful.)

**Comment by jacobdgm** [comment]
Okay, I'm going to open a PR hiding both dropdowns from the Source Detail page in for sources in the Bower segment. I'll also open an Issue, marked as an "enhancement", saying we might consider supporting the display of Sequences on the Browse Chant page - if we end up implementing that, we can reintroduce the "Select a folio" dropdown.

**Comment by jacobdgm** [comment]
Looking into this a bit further, I see that this has nothing to do with non-standard folios - in `get_context_data` for our `ChantListView`, we raise an Http404 error if the source is not in the Cantus segment:

https://github.com/DDMAL/CantusDB/blob/48a6ee50959409d6e8f6b594ed206137cb11f637/django/cantusdb_project/main_app/views/chant.py#L363-L375

Looking back, I was the one who added this code in June 23 - as part of that #785, the "View all chants" link was hidden except in the Cantus segment, but I evidently didn't realize that the dropdowns also linked to the Chant List page. So, by removing these dropdowns for Bower sources, we can complete what #785 was meant to accomplish.
